<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Practical 1</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NORD_practicals</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/markravinet/NORD_practicals">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Practical 1</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NORD_practicals/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommarkravinetNORDpracticalstreeaa987c4380fc05e68fae5be4ffb8ac7d7a71139ctargetblankaa987c4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/markravinet/NORD_practicals/tree/aa987c4380fc05e68fae5be4ffb8ac7d7a71139c" target="_blank">aa987c4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommarkravinetNORDpracticalstreeaa987c4380fc05e68fae5be4ffb8ac7d7a71139ctargetblankaa987c4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/markravinet/NORD_practicals/tree/aa987c4380fc05e68fae5be4ffb8ac7d7a71139c" target="_blank">aa987c4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  code/calc_haplotype_stats.R
    Untracked:  code/identify_candidate_genes.R
    Untracked:  code/plot_pca.R
    Untracked:  code/plot_sliding_window.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/practical1.Rmd</code>) and HTML (<code>docs/practical1.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/markravinet/NORD_practicals/blob/aa987c4380fc05e68fae5be4ffb8ac7d7a71139c/analysis/practical1.Rmd" target="_blank">aa987c4</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-23
</td>
<td>
Fix file typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/markravinet/NORD_practicals/09982ee3bf77cdfee090aa48f5e917efed6b189f/docs/practical1.html" target="_blank">09982ee</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/markravinet/NORD_practicals/1f3a6d1ac0a01ad1931c41aec52f33d74c7a15c7/docs/practical1.html" target="_blank">1f3a6d1</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/markravinet/NORD_practicals/blob/99d5a1fc6f5706363a132b4996ebc6925cc6fb5c/analysis/practical1.Rmd" target="_blank">99d5a1f</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-23
</td>
<td>
Fix errors
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/markravinet/NORD_practicals/423cd16208e692a9813f73258f001fad7edcc30b/docs/practical1.html" target="_blank">423cd16</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/markravinet/NORD_practicals/blob/bb16688cd85cf974eeb49970daf18b52e2eb13bb/analysis/practical1.Rmd" target="_blank">bb16688</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-20
</td>
<td>
Publish initial files for tutorials
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="inferring-selection-in-the-genome---part-1" class="section level1">
<h1>Inferring selection in the genome - part 1</h1>
<p>There are many different ways to detect regions under divergent selection. In this tutorial, we are going to compute three of them in sliding genomic windows: - pi, a measure of genetic variation - <em>F</em><sub>ST</sub> - a measure of genomic differentiation - <em>d</em><sub>XY</sub> - a measure of absolute divergence</p>
<p>Note that dxy <em>d</em><sub>XY</sub> and pi require monomorphic sites to be present in the dataset, whereas <em>F</em><sub>ST</sub> is only computed on bi-allelic sites. It is thus important to filter out indels and multi-allelic sites and to keep monomorphic sites (i.e. do not apply an maf filter).</p>
<p>As SNP <em>F</em><sub>ST</sub> values are very noisy, it is also better to compute <em>F</em><sub>ST</sub> estimates for entire regions. Selection is expected to not only affect a single SNP and the power to detect a selective sweep is thus higher for genomic regions. How large the genomic region should be depends on the SNP density, how fast linkage disequilibrium decays, how recent the sweep is and other factors. It is thus advisable to try different window sizes. Here, we will use 100 kb windows. An alternative is to use windows of fixed number of sites instead of fixed size.</p>
<div id="setting-up" class="section level3">
<h3>Setting up</h3>
<p>In order to calculate these statistics, we will use a set of custom <code>python</code> scripts written by <a href="https://simonmartinlab.org/">Simon Martin</a> which you can download <a href="https://github.com/simonhmartin/genomics_general">here</a>. We will spend a few moments learning how to install these scripts as we will use them here and also in the practical session later today too. Thankfully, they are very easy to set up, we can download and install them using <code>git</code>. We do this like so:</p>
<pre class="shell"><code># move to home directory
cd ~
# use git clone to clone the genomics_general github repo
git clone https://github.com/simonhmartin/genomics_general.git</code></pre>
<p>This will download the scripts into a folder in your home directory. You can look inside to see all the different scripts available. However, what we really want to do is to make sure that they are accessible wherever we are on the cluster. We can do that easily too. We simply add the directory of the scripts to the <code>$PATH</code> variable like so:</p>
<pre class="shell"><code># create a custom path
export PATH=$PATH:$HOME/genomics_general</code></pre>
<p>Now if you try typing the following into the shell, you should see the help dialogue:</p>
<pre class="shell"><code>popgenWindows.py --help</code></pre>
<p>However, the <code>$PATH</code> variable we created just now is temporary. If you log off, it will be gone. So we can echo it to our <code>.bashrc</code> file to make sure it is loaded each time we start a new shell session. We do that like so</p>
<pre class="shell"><code>echo &quot;# create a custom path&quot; &gt; ~/.bashrc
echo &quot;export PATH=$PATH:$HOME/genomics_general:$HOME/genomics_general/VCF_processing&quot; &gt; ~/.bashrc</code></pre>
<p>Now the scripts are easily callable!</p>
</div>
<div id="converting-the-input-files" class="section level3">
<h3>Converting the input files</h3>
<p>First, we will convert a <code>vcf</code> file into Simon Martin’s <code>geno</code> file format. To do that we need to use the <code>parseVCF.py</code> script. The <code>geno</code> file is very straightforward - it is essentially a genotype matrix and easy to create.</p>
<p>We will convert a vcf with data from 105 individuals into a geno file that we can process downstream. First lets set up a directory to work in:</p>
<pre class="shell"><code>cd ~
mkdir genome_scan
cd genome_scan</code></pre>
<p>Then we copy over our data into this directory we have just created:</p>
<pre class="shell"><code>cp /home/example_data/session_08/sparrows_chr8_subset.vcf.gz ./</code></pre>
<p>Before we go further with converting the file, it is worth looking at it in a bit more detail to get some idea of what it contains:</p>
<pre class="shell"><code># what individuals?
bcftools query -l sparrows_chr8_subset.vcf.gz
# how many individuals?
bcftools query -l sparrows_chr8_subset.vcf.gz | wc -l
# how many variants?
bcftools index sparrows_chr8_subset.vcf.gz
bcftools index -n sparrows_chr8_subset.vcf.gz</code></pre>
<p>We can very easily convert the file. We do so using the <code>parseVCF.py</code> script.</p>
<pre class="shell"><code>parseVCF.py -i sparrows_chr8_subset.vcf.gz -o sparrows_chr8_subset.geno</code></pre>
<p>This should take a few seconds to run and then we can have a look at the input:</p>
<pre class="shell"><code>head sparrows_chr8_subset.geno | cut -f 1-10 | column -t</code></pre>
<p>Essentially this is a genotype matrix for each of our individuals. It is also a good idea to compress this file with <code>gzip</code>. This is also really straightforward.</p>
<pre class="shell"><code>gzip sparrows_chr8_subset.geno</code></pre>
<p>Now we are ready to analyse the data!</p>
</div>
<div id="performing-the-analysis" class="section level3">
<h3>Performing the analysis</h3>
<p>We want to calculate <em>d</em><sub>XY</sub>, <em>F</em><sub>ST</sub> and pi from the house sparrow and the bactrianus sparrow from this dataset. We can do this very easily, but first it will help if we create a population file that we can give to the script so that we don’t need to manually input each of the sample names. We can do this fairly easily using <code>grep</code></p>
<pre class="shell"><code># generate all the sample names
bcftools query -l sparrows_chr8_subset.vcf.gz &gt; samples
# get the bactrianus sparrow samples
grep &quot;^[KP][Ad]&quot; samples &gt; bactrianus
# get the house sparrow samples
grep -v &quot;^[KP][Ad]*&quot; samples &gt; house</code></pre>
<p>However, the population file also needs a column with the population denoted in it. So we can generate this with <code>awk</code>.</p>
<pre class="shell"><code>awk &#39;{print $1&quot;\tbac&quot;}&#39; bactrianus &gt; pop_file
awk &#39;{print $1&quot;\thouse&quot;}&#39; house &gt;&gt; pop_file
# now look at the pop_file
cat pop_file</code></pre>
<p>With this, we are ready to perform our analyses.</p>
<pre class="shell"><code>popgenWindows.py -g sparrows_chr8_subset.geno.gz -o div_stats.csv \
   -f phased -w 100000 -m 10 -s 25000 \
   -p house -p bac \
   --popsFile pop_file \
   --writeFailedWindow</code></pre>
<p>Some notes on the options here:</p>
<ul>
<li><code>-g</code> the input geno file</li>
<li><code>-o</code> the output stats file</li>
<li><code>-f</code> the format of the input file - here phased simply means we have both sites (i.e. it is diploid)</li>
<li><code>-w</code> the size of the window in bp</li>
<li><code>-m</code> the minimum number of sites within a window - here set to 10 to get an output</li>
<li><code>-s</code> the size of the step for the sliding window - here it is 25 Kb.</li>
<li><code>-p</code> the population names to compare from the pops file.</li>
<li><code>--popsFile</code> the name of the pops file</li>
<li><code>--writeFailedWindow</code> write to the output even if the window does not meet filtering requirements</li>
</ul>
<p>This will run very quickly and write out our results after printing a dialogue to the screen. We can then use <code>head</code> to examine the results. Of course it is worth remembering here that this is just a dummy dataset that we used to run the analyses quickly. In the next section, when we visualise this in <code>R</code>, we will use a real dataset!</p>
</div>
<div id="visualising-the-results" class="section level3">
<h3>Visualising the results</h3>
<p>The real analysis we will visualise in this step was conducted on 413 individuals from across the <em>Passer</em> sparrow distribution. This includes different species but here we will use a comparison of <em>d</em><sub>XY</sub>, <em>F</em><sub>ST</sub> and pi from house sparrows in Europe versus bactrianus sparrows from Kazahkstan and Iran.</p>
<p>We can get the real data like so:</p>
<pre class="shell"><code>cd genome_scan
cp /home/example_data/session_08/house_bac_div_stats.csv .</code></pre>
<p>Now we will turn to <code>R</code> to visualise our data. The first thing we need to do is prepare the <code>R</code> environment.</p>
<pre class="r"><code># clear the r environment
rm(list = ls())
# load the tidyverse library
library(tidyverse)</code></pre>
<p>Then with this done, we can read in the data. <strong>Again remember you may need to change the path to the files you are reading in depending on where you are running R</strong></p>
<pre class="r"><code>infile &lt;- &quot;./house_bac_div_stats.csv&quot;
house_bac &lt;- read_csv(infile)</code></pre>
<p>We can now take a look at the <code>house_bac</code> dataframe in a bit more detail. Note that it is a <code>tibble</code> so that we can easily get a quick summary of the data. This contains estimates of pi, <em>d</em><sub>XY</sub> and <em>F</em><sub>ST</sub> in 100Kb windows with a 25Kb step along our chromosome of interest (chromosome 8 here).</p>
<p>Let’s plot our statistics to get a better idea of what is going on here. We will start with <em>F</em><sub>ST</sub>:</p>
<pre class="r"><code># fst
ggplot(house_bac, aes(mid, fst_bac_house)) + geom_line()</code></pre>
<p>Next we will look at <em>d</em><sub>XY</sub></p>
<pre class="r"><code># dxy
ggplot(house_bac, aes(mid, dxy_bac_house)) + geom_line()</code></pre>
<p>Clearly there is something going on at the centre of the chromosome here. We should also look at the estimates of pi to see if there is anything we can learn from that too. Since we want to plot pi from both species on the same plot, it requires a bit more leg work. First we need to use the <code>gather</code> function to reshape the data so that we can plot it.</p>
<pre class="r"><code># pi
# first gather the data
pi &lt;- house_bac %&gt;%
  select(mid, contains(&quot;pi&quot;)) %&gt;%
  pivot_longer(cols = contains(&quot;pi&quot;), names_to = &quot;species&quot;, values_to = &quot;pi&quot;)
# then plot it
ggplot(pi, aes(mid, pi, colour = species)) + geom_line()</code></pre>
<p>pi also shows a large drop in the level of diversity at the middle of the chromosome. This suggests an area of reduced recombination, most likely caused by the presence of the centromere. This can also explain the largest <em>F</em><sub>ST</sub> peak present in the data. This suggests we should try an alternative approach.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7       whisker_0.4      knitr_1.36       magrittr_2.0.1  
 [5] R6_2.5.1         rlang_0.4.12     fastmap_1.1.0    fansi_0.5.0     
 [9] stringr_1.4.0    tools_4.1.2      xfun_0.28        utf8_1.2.2      
[13] git2r_0.28.0     jquerylib_0.1.4  htmltools_0.5.2  ellipsis_0.3.2  
[17] rprojroot_2.0.2  yaml_2.2.1       digest_0.6.28    tibble_3.1.5    
[21] lifecycle_1.0.1  crayon_1.4.2     later_1.3.0      vctrs_0.3.8     
[25] promises_1.2.0.1 fs_1.5.0         glue_1.5.0       evaluate_0.14   
[29] rmarkdown_2.11   stringi_1.7.5    compiler_4.1.2   pillar_1.6.4    
[33] httpuv_1.6.3     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
