<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Practical 2</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NORD_practicals</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/markravinet/NORD_practicals">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Practical 2</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NORD_practicals/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220220code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220220)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220220code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220220)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommarkravinetNORDpracticalstreeaa987c4380fc05e68fae5be4ffb8ac7d7a71139ctargetblankaa987c4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/markravinet/NORD_practicals/tree/aa987c4380fc05e68fae5be4ffb8ac7d7a71139c" target="_blank">aa987c4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommarkravinetNORDpracticalstreeaa987c4380fc05e68fae5be4ffb8ac7d7a71139ctargetblankaa987c4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/markravinet/NORD_practicals/tree/aa987c4380fc05e68fae5be4ffb8ac7d7a71139c" target="_blank">aa987c4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  code/calc_haplotype_stats.R
    Untracked:  code/identify_candidate_genes.R
    Untracked:  code/plot_pca.R
    Untracked:  code/plot_sliding_window.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/practical2.Rmd</code>) and HTML (<code>docs/practical2.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/markravinet/NORD_practicals/blob/aa987c4380fc05e68fae5be4ffb8ac7d7a71139c/analysis/practical2.Rmd" target="_blank">aa987c4</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-23
</td>
<td>
Fix file typos
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/markravinet/NORD_practicals/09982ee3bf77cdfee090aa48f5e917efed6b189f/docs/practical2.html" target="_blank">09982ee</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/markravinet/NORD_practicals/1f3a6d1ac0a01ad1931c41aec52f33d74c7a15c7/docs/practical2.html" target="_blank">1f3a6d1</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/markravinet/NORD_practicals/blob/99d5a1fc6f5706363a132b4996ebc6925cc6fb5c/analysis/practical2.Rmd" target="_blank">99d5a1f</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-23
</td>
<td>
Fix errors
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/markravinet/NORD_practicals/423cd16208e692a9813f73258f001fad7edcc30b/docs/practical2.html" target="_blank">423cd16</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/markravinet/NORD_practicals/blob/bb16688cd85cf974eeb49970daf18b52e2eb13bb/analysis/practical2.Rmd" target="_blank">bb16688</a>
</td>
<td>
markravinet
</td>
<td>
2022-02-20
</td>
<td>
Publish initial files for tutorials
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="inferring-selection-in-the-genome---part-2" class="section level1">
<h1>Inferring selection in the genome - part 2</h1>
<div id="getting-access-to-a-larger-dataset" class="section level4">
<h4>Getting access to a larger dataset</h4>
<p>Remember that above, we performed our phasing on a drastically reduced dataset in order to complete our analysis in time. In order to get the real dataset, we should do the following.</p>
<pre class="shell"><code>cd ~
mkdir rehh
cd rehh
cp /home/example_data/session_08/[hb]*.vcf.gz .</code></pre>
<p>These are the full vcfs for the house and bactrianus datasets that we can read into <code>R</code> to perform our genome scan with.</p>
</div>
<div id="running-an-analysis-with-rehh" class="section level3">
<h3>Running an analysis with rehh</h3>
<p>To compute and detect regions with extended haplotype lengths, we will use the excellent <code>rehh</code> R package. For more information see the <a href="https://cran.r-project.org/web/packages/rehh/vignettes/rehh.html">vignette</a>. <code>rehh</code> is well maintained, continually updated and has <a href="https://cran.r-project.org/web/packages/rehh/vignettes/rehh.html">a very informative tutorial</a> which I recommend you also check out.</p>
<p>To run <code>rehh</code> and perform our analyses, we need to run things in <code>R</code>. You can either download the data from <a href="https://github.com/speciationgenomics">our github</a> and run it locally on your own machine, or we can use our <code>RStudio</code> server. Once <code>R</code> is running, we are ready to go!</p>
<div id="setting-up-the-r-environment" class="section level4">
<h4>Setting up the R environment</h4>
<p>The first thing we need to do is clear our <code>R</code> environment and load the packages we need. Like so:</p>
<pre class="r"><code># clear environment
rm(list = ls())
# load packages
library(rehh)
library(tidyverse)</code></pre>
</div>
</div>
<div id="reading-in-data-from-vcfs" class="section level3">
<h3>Reading in data from vcfs</h3>
<p>A nice new addition to <code>rehh</code> is the ability to read in (and also filter) your data from a vcf. However, it is still quite tricky to split up individuals so instead we will read in a vcf for each population. We read in data using the <code>data2haplohh</code> function. Remember again, <strong>you may need to change the path to these files depending on where you have placed them</strong></p>
<pre class="r"><code># read in data for each species
# house
house_hh &lt;- data2haplohh(hap_file = &quot;./rehh/house_chr8.vcf.gz&quot;,
                   polarize_vcf = FALSE)
# bactrianus
bac_hh &lt;- data2haplohh(hap_file = &quot;./rehh/bac_chr8.vcf.gz&quot;,
                         polarize_vcf = FALSE)</code></pre>
<p>This will take a few moments but once the commands are run, we will have read in our data. It is really important that we set <code>polarize_vcf</code> to <code>FALSE</code> because we have not used an outgroup genome to set our alleles as derived or ancestral. <strong>NB. Some plots and functions in <code>rehh</code> will still refer to ancestral and derived even if you use this option</strong>. Instead <code>rehh</code> will use the minor and major alleles (in terms of frequency) from our data.</p>
<p>Next, we will filter our data on a <strong>minor allele frequency</strong> or <strong>MAF</strong>. This is really simple in <code>rehh</code> with the <code>subset</code> function:</p>
<pre class="r"><code># filter on MAF - here 0.05
house_hh_f &lt;- subset(house_hh, min_maf = 0.05)
bac_hh_f &lt;- subset(bac_hh, min_maf = 0.05)</code></pre>
<p>This will remove some sites - and we’ll be ready to run our haplotype scans.</p>
</div>
<div id="performing-a-haplotype-genome-scan---xpehh" class="section level3">
<h3>Performing a haplotype genome scan - <em>xpEHH</em></h3>
<p>Before we can calculate the statistic we are interested in <em>xpEHH</em> - we need to calculate <em>iES</em> statistics. Luckily this is really easy using the <code>rehh</code> function <code>scan_hh</code>.</p>
<pre class="r"><code># perform scans
house_scan &lt;- scan_hh(house_hh_f, polarized = FALSE)
bac_scan &lt;- scan_hh(bac_hh_f, polarized = FALSE)</code></pre>
<p>Next we will calculate <em>xpEHH</em> which is the cross-population <em>EHH</em> test. This is essentially a test for the probability that if we randomly sampled haplotypes from different populations, we would get different haplotypes. Again, <code>rehh</code> makes this simple with the <code>ies2xpehh</code> function.</p>
<pre class="r"><code># perform xp-ehh
house_bac &lt;- ies2xpehh(bac_scan, house_scan,
                       popname1 = &quot;bactrianus&quot;, popname2 = &quot;house&quot;,
                       include_freq = T)</code></pre>
<p>Here we provide the names of our previous <em>iES</em> scans (<code>bac_scan</code> and <code>house_scan</code>). We can also provide the function with the names of our populations and finally, if we set <code>include_freq</code> to <code>TRUE</code>, we get the frequencies of alleles in our output, which might be useful if we want to see how selection is acting on a particular position.</p>
<p>Next, we can plot the <em>xpEHH</em> values, like so:</p>
<pre class="r"><code># plot
ggplot(house_bac, aes(POSITION, XPEHH_bactrianus_house)) + geom_point()</code></pre>
<p>In this plot, highly negative values suggest selection in population 2 (house in this case) whereas positive values indicate selection in population 1. Alternatively, like with <em>iHS</em>, we could plot the log <em>P</em> values.</p>
<pre class="r"><code>ggplot(house_bac, aes(POSITION, LOGPVALUE)) + geom_point()</code></pre>
</div>
<div id="examining-haplotype-structure-around-a-target-of-selection" class="section level3">
<h3>Examining haplotype structure around a target of selection</h3>
<p>One other nice feature of <code>rehh</code> is that we can examine haplotype structure around SNPs we think might be under selection. Before we do that, we need to identify the SNP in our dataset with the strongest evidence of being an <em>xpEHH</em> outlier.</p>
<pre class="r"><code># find the highest hit
hit &lt;- house_bac %&gt;% arrange(desc(LOGPVALUE)) %&gt;% top_n(1)
# get SNP position
x &lt;- hit$position</code></pre>
<p>Here we also set the position of our putative selection SNP as the object <code>x</code>. This is because we need to identify where it occurs in our haplotype objects - unfortunately we cannot use the position for this. In the code below, we find the marker id for both our datasets.</p>
<pre class="r"><code>marker_id_h &lt;- which(house_hh_f@positions == x)
marker_id_b &lt;- which(bac_hh_f@positions == x)</code></pre>
<p>Now we are ready to plot the bifurcation of haplotypes around our site of selection. We do this like so:</p>
<pre class="r"><code>house_furcation &lt;- calc_furcation(house_hh_f, mrk = marker_id_h)
bac_furcation &lt;- calc_furcation(bac_hh_f, mrk = marker_id_b)</code></pre>
<p>We can also plot both of these to have a look at them:</p>
<pre class="r"><code>plot(house_furcation, xlim = c(19.18E+6, 19.22E+6))
plot(bac_furcation, xlim = c(19.18E+6, 19.22E+6))</code></pre>
<p>Calculating the furcation pattern also makes it possible to calculate the haplotype length around our signature of selection.</p>
<pre class="r"><code>house_haplen &lt;- calc_haplen(house_furcation)
bac_haplen &lt;- calc_haplen(bac_furcation)</code></pre>
<p>With the haplotype length calculated, we can now plot this to see how haplotype structure differs between our two populations.</p>
<pre class="r"><code>plot(house_haplen)
plot(bac_haplen)</code></pre>
<p>Here we can see the blue haplotype is much larger around this target and is also more numerous in the European house sparrow.</p>
</div>
<div id="writing-out-the-data-for-later-used" class="section level3">
<h3>Writing out the data for later used</h3>
<p>Finally, before we move on to the last tutorial, we are going to write out the data. We’ll also make the column names all smaller letters, to make downstream scripting a bit easier.</p>
<p>Use the following code to achieve this:</p>
<pre class="r"><code># write out house bactrianus xpEHH
house_bac &lt;- tbl_df(house_bac)
colnames(house_bac) &lt;- tolower(colnames(house_bac))
write_tsv(house_bac, &quot;./house_bac_xpEHH.tsv&quot;)</code></pre>
<p>In the last tutorial, we’ll use <code>R</code> to identify genes that are close to our outlier SNPs.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7       whisker_0.4      knitr_1.36       magrittr_2.0.1  
 [5] R6_2.5.1         rlang_0.4.12     fastmap_1.1.0    fansi_0.5.0     
 [9] stringr_1.4.0    tools_4.1.2      xfun_0.28        utf8_1.2.2      
[13] git2r_0.28.0     jquerylib_0.1.4  htmltools_0.5.2  ellipsis_0.3.2  
[17] rprojroot_2.0.2  yaml_2.2.1       digest_0.6.28    tibble_3.1.5    
[21] lifecycle_1.0.1  crayon_1.4.2     later_1.3.0      vctrs_0.3.8     
[25] promises_1.2.0.1 fs_1.5.0         glue_1.5.0       evaluate_0.14   
[29] rmarkdown_2.11   stringi_1.7.5    compiler_4.1.2   pillar_1.6.4    
[33] httpuv_1.6.3     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
